composer/trainer/trainer.py

```diff
@@ -1022,9 +1100,157 @@ def saved_checkpoints(self) -> List[Tuple[Timestamp, List[pathlib.Path]]]:
+        if self._train_data_spec is None:
+            _raise_missing_argument_exception("train_dataloader")
+        if train_subset_num_batches is not None:
+            self.state.dataloader_len = train_subset_num_batches
+        if compute_training_metrics is not None:
+            self.train_metrics = _get_training_metrics(self._original_model) if compute_training_metrics else None
+
+        # Max Duration
+        if max_duration is not None:
+            self.state.max_duration = ensure_time(max_duration)
+
+        if self.state.max_duration is None:
+            _raise_missing_argument_exception("max_duration")
+
+        # Scale Schedule Ratio and Schedulers
+        if scale_schedule_ratio != 1.0:
+            self.state.max_duration = _scale_max_duration_by_ssr(scale_schedule_ratio, self.state.max_duration)
+        if schedulers is not None:
+            self.state.schedulers = _compile_schedulers(schedulers, self.state, scale_schedule_ratio)
+            self._step_schedulers_every_batch = _should_step_schedulers_every_batch(schedulers,
+                                                                                    step_schedulers_every_batch)
+        else:
+            if scale_schedule_ratio != 1.0:
+                warnings.warn("Specifying `scale_schedule_ratio` without `schedulers` has no effect.")
+
+            if step_schedulers_every_batch is not None:
+                warnings.warn("Specifying `step_schedulers_every_batch` without `schedulers` has no effect.")
+
+            if step_schedulers_every_batch is not None:
+                warnings.warn("Specifying `step_schedulers_every_batch` without `schedulers` has no effect.")
+
+        if len(ensure_tuple(schedulers)) == 0:
+            warnings.warn(f"NoSchedulerWarning: No schedulers were specified. The learning rate will be constant.")
+
+        # Evaluators
+        if eval_dataloader is not None:
+            self.evaluators = [
+                ensure_evaluator(evaluator, self._original_model.metrics(train=False))
+                for evaluator in ensure_tuple(eval_dataloader)
+            ]
+            _set_evaluator_interval_and_subset_num_batches(
+                evaluators=self.evaluators,
+                eval_interval=eval_interval,
+                subset_num_batches=eval_subset_num_batches,
+            )
+            if len(self.evaluators) == 0:
+                if eval_subset_num_batches != -1:
+                    warnings.warn("Specifying `eval_subset_num_batches` without an `eval_dataloader` has no effect.")
+                if eval_interval != 1:
+                    warnings.warn("Specifying `eval_interval` without an `eval_dataloader` has no effect.")
+
+        if len(self.evaluators) == 0:
+            warnings.warn(("No `eval_dataloader` was specified. Please specify `eval_dataloader` to periodically "
+                           "evaluate your model while training."))
+
+        # Grad Accum
+        if grad_accum is not None:
```

Comment:
for readability, and to downscope the helper functions, I wonder if this should be:
```
if grad_accum == 'auto':
     self.state.grad_accum, self.adaptive_gradient_accumulation = _configure_adaptive_ga(grad_accum, self._device)
elif grad_accum is not None:
    self.state.grad_accum = grad_accum
```

```

Reply:
Refactored into two helper functions -- `_is_adaptive_grad_accum` and `_get_initial_grad_accum`